File: Kotlin/src/schedular.kt
import java.time.LocalTime
import java.time.format.DateTimeFormatter
import java.time.format.FormatStyle
import java.time.Duration

class Meeting {
  
  val meeting = this
  val starts = this
  val and = this
  val add = this
  val participants = this
  var event_name = ""
  var start_time = ""
  var end_time = ""
  var date = ""
  var participant_list = listOf<String>()

  infix fun name(event_name: String): Unit {
    this.event_name = event_name
  }
  
  fun format(time: Double): String {
	return "%.2f".format(time).replace(".", ":")
  }

  infix fun at(start_time: Double): Unit {
    this.start_time = format(start_time)
  }

  infix fun ends(end_time: Double): Unit {
    this.end_time = format(end_time)
  }

  infix fun include(people: String): Unit {
    this.participant_list = people.dropLast(1).split(".")
  }

  fun duration(): String {
    val t1 = LocalTime.parse(this.start_time, DateTimeFormatter.ofPattern("HH:mm"))
    val t2 = LocalTime.parse(this.end_time, DateTimeFormatter.ofPattern("HH:mm"))
    var minutes = Duration.between(t1,t2).toMinutes()
    return LocalTime.MIN.plus(Duration.ofMinutes(minutes)).toString()
  }

  infix fun Int.January(year: Int) = "1/$this/$year"
  infix fun Int.February(year: Int) = "2/$this/$year"
  infix fun Int.March(year: Int) = "3/$this/$year"
  infix fun Int.April(year: Int) = "4/$this/$year"
  infix fun Int.May(year: Int) = "5/$this/$year"
  infix fun Int.June(year: Int) = "6/$this/$year"
  infix fun Int.July(year: Int) = "7/$this/$year"
  infix fun Int.August(year: Int) = "8/$this/$year"
  infix fun Int.September(year: Int) = "9/$this/$year"
  infix fun Int.October(year: Int) = "10/$this/$year"
  infix fun Int.November(year: Int) = "11/$this/$year"
  infix fun Int.December(year: Int) = "12/$this/$year"
}

object schedule {
  infix fun meeting(block: Meeting.() -> Unit): Meeting {
    return Meeting().apply(block)
  }
}

File: Ruby/src/schedular.rb
class Meeting
  attr_reader :meeting_name, :start_time, :end_time, :date, :participant_list

  def name(meeting_name)
    @meeting_name = meeting_name
  end

  def convert_time(time)
    ('%.2f' % time).sub('.', ':')
  end

  def start(start_time)
    @start_time = convert_time(start_time)
  end

  def finish(end_time)
    @end_time = convert_time(end_time)
  end

  def on(month, day, year)
    @date = "#{month}/#{day}/#{year}"
  end

  def participants(participants)
    @participant_list = participants
  end

  def meeting_duration()
    date_split = @date.split('/')

    start_split = @start_time.to_s.split(":")
    start = Time.new(date_split[2], date_split[0], date_split[1], start_split[0], start_split[1])

    finish_split = @end_time.to_s.split(":")
    finish = Time.new(date_split[2], date_split[0], date_split[1], finish_split[0], finish_split[1])

    Time.at(finish - start).utc.strftime("%H:%M")
  end
end

def schedule(event)
  event
end

def meeting(&block) 
  scheduled_meeting = Meeting.new

  scheduled_meeting.instance_eval(&block)

  scheduled_meeting
end
